app/components/candidate_interface/rejection_reasons_component.rb

```diff
@@ -52,5 +66,13 @@ def rejection_reasons_row(application_choice)
         }
       end
     end
+
+    def rejected_application_choices
+      @rejected_application_choices ||= begin
+        rejected_applications = @application_form.application_choices.includes(:course, :provider, [:offered_course_option]).rejected
```

Comment:
Have `application_choices` been 'included' at this point?

Reply:
As far as I can see, no they've not been pre-loaded by an `include`. It looks like the object that ends up in `@application_form` is the `current_application.previous_application_form`. So it looks like the query that loads `application_choices` is a necessary one-off. (There is only a single instance of the `RejectionReasonsComponent`).